File: SmartPay/Models/Merchant.cs
namespace SmartPay.Models;

public class Merchant
{
    public int Id { get; set; }
    
    public MerchantCategory? Category { get; set; }
    public List<Product> Products { get; set; }
    public List<User> Users { get; set; }
    
    public string Name { get; set; }
}

public class MerchantViewModel
{
    public int Id { get; set; }

    public string Name { get; set; }
}

File: SmartPay/ClientApp/src/api/models/Merchant.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { MerchantCategory } from './MerchantCategory';
import type { Product } from './Product';
import type { User } from './User';

export type Merchant = {
    id?: number;
    category?: MerchantCategory;
    products?: Array<Product> | null;
    users?: Array<User> | null;
    name?: string | null;
};